clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformerUtility.java

```diff
@@ -30,6 +30,12 @@ public class DaVinciRecordTransformerUtility<K, O> {
 
   public DaVinciRecordTransformerUtility(DaVinciRecordTransformer recordTransformer) {
     this.recordTransformer = recordTransformer;
+
+    Schema keySchema = recordTransformer.getKeySchema();
+    Schema outputValueSchema = recordTransformer.getOutputValueSchema();
+    this.keyDeserializer = new AvroGenericDeserializer<>(keySchema, keySchema);
+    this.outputValueDeserializer = new AvroGenericDeserializer<>(outputValueSchema, outputValueSchema);
+    this.outputValueSerializer = new AvroSerializer<>(outputValueSchema);
```

Comment:
I think these properties can become final, now that they're set in the constructor rather than lazily, right?